core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -507,17 +525,25 @@ public void stopPartitions(Set<StopPartition> stopPartitions,
                 LOGGER.error("Error while stopping the partition: {}", stopPartition, ex);
             }
         }
-        // Note `deleteLocalLog` will always be true when `deleteRemoteLog` is true but not the other way around.
         Set<TopicIdPartition> deleteLocalPartitions = stopPartitions.stream()
                 .filter(sp -> sp.deleteLocalLog() && topicIdByPartitionMap.containsKey(sp.topicPartition()))
                 .map(sp -> new TopicIdPartition(topicIdByPartitionMap.get(sp.topicPartition()), sp.topicPartition()))
                 .collect(Collectors.toSet());
+
         if (!deleteLocalPartitions.isEmpty()) {
-            // NOTE: In ZK mode, this#stopPartitions method is called when Replica state changes to Offline and
-            // ReplicaDeletionStarted
-            remoteLogMetadataManager.onStopPartitions(deleteLocalPartitions);
             deleteLocalPartitions.forEach(tpId -> topicIdByPartitionMap.remove(tpId.topicPartition()));
         }
+
+        // NOTE: In ZK mode, this#stopPartitions method is called when Replica state changes to Offline and
+        // ReplicaDeletionStarted
+        Set<TopicIdPartition> stopRLMMPartitions = stopPartitions.stream()
```

Comment:
can we merge L539 and L528 together?

Reply:
Do you mean this:
```
Set<TopicIdPartition> pendingActionsPartitions = stopPartitions.stream()
                .filter(sp -> (sp.stopRemoteLogMetadataManager() || sp.deleteLocalLog()) && topicIdByPartitionMap.containsKey(sp.topicPartition()))
                .map(sp -> new TopicIdPartition(topicIdByPartitionMap.get(sp.topicPartition()), sp.topicPartition()))
                .collect(Collectors.toSet());

if (!pendingActionsPartitions.isEmpty()) {
  pendingActionsPartitions.forEach(tpId -> topicIdByPartitionMap.remove(tpId.topicPartition()));
  remoteLogMetadataManager.onStopPartitions(pendingActionsPartitions)'
}
```

I'm not sure if it's correct. when `sp.stopRemoteLogMetadataManager` is true, the `sp.deleteLocalLog()` could be false because we set `remote.copy.disabled=true` only. In this case, we can't remote `topicIdByPartitionMap`, right?

Reply:
Check and think again. You are totally right! Thanks for the explanation! Updated.